src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java

```diff
@@ -765,56 +735,35 @@ public synchronized void sync() throws BackingStoreException {
             shared = !isSystemRootWritable;
         }
         synchronized (isUserNode()? userLockFile:systemLockFile) {
-           if (!lockFile(shared))
-               throw(new BackingStoreException("Couldn't get file lock."));
-           final Long newModTime =
-                AccessController.doPrivileged(
-                    new PrivilegedAction<Long>() {
-               public Long run() {
-                   long nmt;
-                   if (isUserNode()) {
-                       nmt = userRootModFile.lastModified();
-                       isUserRootModified = userRootModTime == nmt;
-                   } else {
-                       nmt = systemRootModFile.lastModified();
-                       isSystemRootModified = systemRootModTime == nmt;
-                   }
-                   return nmt;
-               }
-           });
+           if (!lockFile(shared)) {
+               throw (new BackingStoreException("Couldn't get file lock."));
+           }
+           long nmt;
+           if (isUserNode()) {
+               nmt = userRootModFile.lastModified();
+               isUserRootModified = userRootModTime == nmt;
+           } else {
+               nmt = systemRootModFile.lastModified();
+               isSystemRootModified = systemRootModTime == nmt;
+           }
+           final long newModTime = nmt;
            try {
                super.sync();
-               AccessController.doPrivileged(new PrivilegedAction<Void>() {
-                   public Void run() {
-                   if (isUserNode()) {
-                       userRootModTime = newModTime.longValue() + 1000;
-                       userRootModFile.setLastModified(userRootModTime);
-                   } else {
-                       systemRootModTime = newModTime.longValue() + 1000;
-                       systemRootModFile.setLastModified(systemRootModTime);
-                   }
-                   return null;
-                   }
-               });
+               if (isUserNode()) {
+                   userRootModTime = newModTime + 1000;
+                   userRootModFile.setLastModified(userRootModTime);
+               } else {
+                   systemRootModTime = newModTime + 1000;
+                   systemRootModFile.setLastModified(systemRootModTime);
+               }
            } finally {
                 unlockFile();
            }
         }
     }
 
-    @SuppressWarnings("removal")
     protected void syncSpi() throws BackingStoreException {
-        try {
-            AccessController.doPrivileged(
-                new PrivilegedExceptionAction<Void>() {
-                public Void run() throws BackingStoreException {
-                    syncSpiPrivileged();
-                    return null;
-                }
-            });
-        } catch (PrivilegedActionException e) {
-            throw (BackingStoreException) e.getException();
-        }
+        syncSpiPrivileged();
```

Comment:
Inline syncSpiPrivileged(), it doesn't need to be separate and the name "privileged" is misleading